Version 1

steven-murray opened the pull request:
This PR implements a bunch of infrastructure changes, including flake8 and black linting, conversion to setuptools_scm for versioning, and slightly better dependency management.

It will take the code to version one (once merged).

steven-murray commented:
@MuellerSeb just pinging you that I've requested your review on this :-)

MuellerSeb commented on .flake8:
```diff
@@ -0,0 +1,38 @@
+    F403
+    F401
+    N803  # I like uppercase single-character names sometimes.
+    N806  # Same here
```
Since it is a rather technical package, I think this is OK.

MuellerSeb commented on .travis.yml:
```diff
@@ -54,13 +53,20 @@ jobs:
+        - python -m coveralls
       name: Coverage on Linux
 
+    - stage: lint
```
I am really curious how this works ;-)

MuellerSeb commented on docs/conf.py:
```diff
@@ -68,27 +70,12 @@
-version = find_version("..","hankel", "__init__.py")
 # The full version, including alpha/beta/rc tags.
-release = find_version("..","hankel", "__init__.py")
+release = get_distribution('hankel').version
```
This seems to be nice and easy. Like it.

MuellerSeb commented on hankel/__init__.py:
```diff
@@ -1,7 +1,4 @@
 """
-hankel API Summary
```
Isn't that needed for the sphinx doc?

MuellerSeb commented on hankel/hankel.py:
```diff
@@ -273,18 +272,16 @@ def xrange_approx(cls, h, nu, k=1):
 
     @classmethod
-    def G(cls, f, h, k=None, *args, **kwargs):
+    def final_term_amplitude(cls, f, h, k=None, *args, **kwargs):
```
I like descriptive naming.

MuellerSeb commented on hankel/hankel.py:
```diff
@@ -298,18 +295,33 @@ def G(cls, f, h, k=None, *args, **kwargs):
         return np.sqrt(np.pi / (2 * h)) * f(np.pi * np.pi / h / k)
 
+    @classmethod
+    def G(cls, f, h, k=None, *args, **kwargs):
```
Why is it still called `G` here and not `final_term_amplitude`?

steven-murray replied:
Good catch. Fixed it up.

MuellerSeb commented on tests/test_get_h.py:
```diff
@@ -113,7 +116,7 @@ def test_xrange():
 
 
-def test_G():
+def test_g():
```
Wasn't that renamed to `final_term_amplitude`?

steven-murray replied:
Yup. Fixed.

MuellerSeb commented:
Another question is, if we really need a deprecation cycle with version 1.0? Regarding the `G` method.

MuellerSeb commented:
I am fine with the changes.
When you are adamant that the `G` method should be in v1 with a deprecation round, I am fine with that.
But one could argue, that everything pre-v1 could be seen as beta versions, where this major jump would allow to make changes like that without a deprecation warning.
But you can decide that, I am fine either way ;-)

steven-murray commented:
I think you're right that in general, going to a new major version allows breaking API. However, since this is the first version, I want to be a bit extra careful. I've added in proper deprecation warnings. From here on, breaking API should be done only on major versions, and only with 2 minor versions notice. 

I also fixed up some docs that were rendering incorrectly.
